src/widgets/modals/DemoUserModal.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Image, StyleSheet, Text, TextInput, View, TouchableOpacity } from 'react-native';
+import { Button } from 'react-native-ui-components';
+import Modal from 'react-native-modalbox';
+import { DemoSiteRequest } from '../../authentication';
+import globalStyles, { SUSSOL_ORANGE, GREY, WARM_GREY } from '../../globalStyles';
+import { authStrings, generalStrings, demoUserModelStrings } from '../../localization';
+import { ConfirmModal } from '../../widgets';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+export class DemoUserModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.demoSiteRequest = new DemoSiteRequest();
+    this.state = {
+      status: 'submit', // submit, submitting, submitted, error
+      error: '',
+      username: '',
+      password: '',
+      repeatPassword: '',
+      email: '',
+    };
+    this.errorTimeoutId = null;
+  }
+
+  componentWillUpdate() {
+    if (this.errorTimeoutId) clearTimeout(this.errorTimeoutId);
+  }
+
+  onDemoRequestSubmit = async () => {
+    if (! this.canAttemptSubmit) return false;
```

Comment:
Space after `!` is a bit weird

Reply:
Resolve with https://github.com/sussol/mobile/pull/724/commits/3344e0da4e4de5e51dffaa570472a34daa5cce5d